File: elixir_phoenix/web/controllers/application_controller.ex
defmodule ElixirPhoenix.ApplicationController do
  use ElixirPhoenix.Web, :controller

  alias ElixirPhoenix.User

  def ping(conn, _param) do
    text conn, "pong"
  end

  def fact(conn, %{ "n" => n }) do
    text conn, Enum.reduce(1..String.to_integer(n), &(&1 * &2))
  end

  def user(conn, %{ "user_id" => user_id }) do
    case Repo.get(User, user_id) do
      nil ->
        conn
        |> put_status(:not_found)
        |> text("")
      user -> render conn, user: user
    end
  end
end

File: ruby_rails/app/controllers/application_controller.rb
class ApplicationController < ActionController::Base
  protect_from_forgery with: :exception

  def ping
    render text: "pong"
  end

  def fact
    render text: calc_fact(params[:n])
  end

  def user_get
    render json: get_user(params[:user_id]).to_json
  end

  private

  def calc_fact(n)
    (1..n.to_i).reduce(:*)
  end

  def get_user(user_id)
    User.find_by(id: user_id.to_i)
  end
end
